pinot-common/src/main/java/org/apache/pinot/common/metadata/ZKMetadataProvider.java

```diff
@@ -73,12 +74,61 @@ public static void setUserConfig(ZkHelixPropertyStore<ZNRecord> propertyStore, S
     propertyStore.set(constructPropertyStorePathForUserConfig(username), znRecord, AccessOption.PERSISTENT);
   }
 
+  @Deprecated
   public static void setTableConfig(ZkHelixPropertyStore<ZNRecord> propertyStore, String tableNameWithType,
       ZNRecord znRecord) {
     propertyStore.set(constructPropertyStorePathForResourceConfig(tableNameWithType), znRecord,
         AccessOption.PERSISTENT);
   }
 
+  /**
+   * Create table config, fail if existed.
+   *
+   * @return true if creation is successful.
+   */
+  public static boolean createTableConfig(ZkHelixPropertyStore<ZNRecord> propertyStore, String tableNameWithType,
```

Comment:
Let's remove `tableNameWithType`, which is just `tableConfig.getTableName()`. Same for the other 2 methods